prow/gerrit/adapter/adapter.go

```diff
@@ -491,16 +504,37 @@ func (c *Controller) processChange(logger logrus.FieldLogger, instance string, c
 		}
 	case client.New:
 		var presubmits []config.Presubmit
+		// Gerrit server might be unavailable intermittently, retry inrepoconfig
+		// processing for increased reliability.
 		for attempt := 0; attempt < inRepoConfigRetries; attempt++ {
 			presubmits, err = c.inRepoConfigCacheHandler.GetPresubmits(trimmedHostPath, func() (string, error) { return baseSHA, nil }, func() (string, error) { return change.CurrentRevision, nil })
 			if err == nil {
 				break
 			}
 		}
-		if err := c.handleInRepoConfigError(err, instance, change); err != nil {
+		if err != nil {
+			// Reports error back to Gerrit. handleInRepoConfigError is
+			// responsible for not sending the same message again and again on
+			// the same commit.
+			if postErr := c.handleInRepoConfigError(err, instance, change); postErr != nil {
+				logger.WithError(postErr).Error("Failed reporting inrepoconfig processing error back to Gerrit.")
+			}
+			// There is no need to keep going when failed to get inrepoconfig
+			// jobs.
+			// Imagining the scenario that:
+			// - Commit #abc triggered static job job-A, inrepoconfig jobs job-B
+			// and job-C
+			// - Both job-B and job-C failed
+			// - Commit #def was pushed. Inrepoconfig failed, falling back to
+			// trigger static job job-A.
+			// - job-A passed.
+			// - Prow would make decision on the result of job-A and ignore the
+			// rest. (Yes this is a Prow bug, which should not be a problem when
+			// each prowjob is reported to an individual Gerrit Check).
+			// So long story short: kicking off partial prowjobs is worse than
+			// kicking off nothing.
 			return err
 		}
-		presubmits = append(presubmits, c.config().PresubmitsStatic[cloneURI.String()]...)
```

Comment:
I don't think we can make this change separately from https://github.com/kubernetes/test-infra/pull/27623. If this PR merged alone we would omit the static jobs because `GetPresubmits` is using the trimmedHostPath not the cloneURI as the identifier and will therefore fail to find the static jobs indexed under the cloneURI.